### PR TITLE
feat: Separate DAG for generating Looker artifacts

### PR DESCRIPTION
**Pull request by scholtzan**
## Description

This moves the lookml generation out of the `probe_scraper` DAG into a new `looker` DAG. Having a separate DAG will enable us to trigger lookml generation from the CI of the lookml-generator repository.

### CODE REVIEW

File: dags/looker.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    secret="airflow-gke-secrets",
+    key="probe_scraper_secret__dataops_looker_github_secret_access_token",
+)
+
+
+with DAG(
+    "looker",
+    doc_md=DOCS,
+    default_args=default_args,
+    schedule_interval=None,
```
Comment (scholtzan) [inline dags/looker.py:82]: This DAG doesn't need to be scheduled. It's either triggered from the `probe_scraper` DAG after schema deploys or via CI

File: dags/probe_scraper.py [hunk 2]
```diff
@@ -394,71 +352,11 @@ def choose_branch(self, context):
...
-        secrets=[
-            looker_repos_secret_git_ssh_key_b64,
-            looker_api_client_id_staging,
-            looker_api_client_secret_staging,
-            dataops_looker_github_secret_access_token,
-        ],
-        **airflow_gke_prod_kwargs,
-    )
-
-    delay_python_task >> lookml_generator_staging >> lookml_generator_prod
```
Comment (BenWu) [inline dags/probe_scraper.py:460]: Should this dependency be maintained?  Looks like `trigger_looker` just runs from the start

File: dags/probe_scraper.py [hunk 3]
```diff
@@ -394,71 +352,11 @@ def choose_branch(self, context):
...
-        secrets=[
-            looker_repos_secret_git_ssh_key_b64,
-            looker_api_client_id_prod,
-            looker_api_client_secret_prod,
-            dataops_looker_github_secret_access_token,
-        ],
-        **airflow_gke_prod_kwargs,
+    # trigger lookml generation
+    trigger_looker = TriggerDagRunOperator(
+        task_id="lookml_generator_staging", trigger_dag_id="looker"
```
Comment (BenWu) [inline dags/probe_scraper.py:357]: I think the task id should be updated as well
```suggestion
        task_id="trigger_lookml_generator", trigger_dag_id="looker"
```

File: dags/probe_scraper.py [hunk 4]
```diff
@@ -478,4 +376,4 @@ def choose_branch(self, context):
         dag=dag,
     )
 
-    lookml_generator_prod >> glean_dictionary_netlify_build
+    trigger_looker >> glean_dictionary_netlify_build
```
Comment (BenWu) [inline dags/probe_scraper.py:379]: Does `TriggerDagRunOperator` succeed immediately after triggering or does it wait for the run status?

**Review by kwindau — APPROVED** [review]

File: dags/probe_scraper.py (hunk 4 above)
Comment (scholtzan) [reply dags/probe_scraper.py:379]: There is a `wait_for_completion` parameter which would wait for the DAG run. I'll add it.
Although Glean dictionary doesn't really depend on anything that is in Looker

Comment (BenWu) [reply dags/probe_scraper.py:379]: Found it https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/trigger_dagrun/index.html

There's a `wait_for_completion` param that defaults to false.  Seems like we want it to be true?  Assuming glean dictionary is actually dependent on lookml, I'm not sure

Comment (scholtzan) [reply dags/probe_scraper.py]: I added it to keep it consistent with the current state

**Pull request merged**